src/tests/test_sphincsplus_fors.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+* (C) 2023 Jack Lloyd
+*     2023 Fabian Albert, René Meusel - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2) || defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE)
+
+   #include <botan/hash.h>
+   #include <botan/hex.h>
+
+   #include <botan/assert.h>
+   #include <botan/sp_parameters.h>
+   #include <botan/internal/loadstor.h>
+   #include <botan/internal/sp_address.h>
+   #include <botan/internal/sp_fors.h>
+   #include <botan/internal/sp_hash.h>
+
+namespace Botan_Tests {
+
+class SPHINCS_Plus_FORS_Test final : public Text_Based_Test {
+   private:
+      static Botan::Sphincs_Address read_address(std::span<const uint8_t> address_buffer) {
+         BOTAN_ASSERT_NOMSG(address_buffer.size() == 32);
+
+         std::array<uint32_t, 8> adrs;
+         for(size_t i = 0; i < 8; ++i) {
+            adrs[i] = Botan::load_be<uint32_t>(address_buffer.data(), i);
+         }
+
+         return Botan::Sphincs_Address(adrs);
+      }
+
+   public:
+      SPHINCS_Plus_FORS_Test() :
+            Text_Based_Test("pubkey/sphincsplus_fors.vec",
+                            "SphincsParameterSet,Address,SecretSeed,PublicSeed,PublicKey,Msg,Signature") {}
+
+      bool skip_this_test(const std::string&, const VarMap& vars) override {
+         [[maybe_unused]] auto params = Botan::Sphincs_Parameters::create(vars.get_req_str("SphincsParameterSet"));
```

Comment:
Again, shouldn't ever really be unused since at least one of the two modes must be available.

Reply:
As with the previous one. If both hash functions are enabled, `params` is unused.